Source/Scene/ColorMapExpression.js

```diff
@@ -109,6 +127,15 @@ define([
     ColorMapExpression.prototype.evaluate = function(feature) {
         var value = feature.getProperty(this._propertyName);
 
+        if (defined(this._pattern)) {
+            var match = this._pattern.exec(value);
+            if (defined(match)) {
+                value = match[1];
```

Comment:
Make a new local variable instead of assigning to `value` again (it will also default to `undefined`).